src/common/common.c

```diff
@@ -233,31 +245,64 @@ OQS_API void OQS_MEM_insecure_free(void *ptr) {
 }
 
 void *OQS_MEM_aligned_alloc(size_t alignment, size_t size) {
-#if defined(__MINGW32__) || defined(__MINGW64__)
-	return __mingw_aligned_malloc(size, alignment);
-#elif defined(_MSC_VER)
-	return _aligned_malloc(size, alignment);
-#elif defined(_ISOC11_SOURCE) // glibc
+#if defined(OQS_HAVE_ALIGNED_ALLOC)  // glibc
 	return aligned_alloc(alignment, size);
-#elif (defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 200112L) || defined(__APPLE__)
+#else
+	// Check alignment (power of 2, and >= sizeof(void*)) and size (multiple of alignment)
+	if (alignment & (alignment - 1) || size & (alignment - 1) || alignment < sizeof(void *)) {
+		errno = EINVAL;
+		return NULL;
+	}
+
+#if defined(OQS_HAVE_POSIX_MEMALIGN)
 	void *ptr = NULL;
 	const int err = posix_memalign(&ptr, alignment, size);
 	if (err) {
 		errno = err;
 		ptr = NULL;
 	}
 	return ptr;
-#else // musl, maybe others.
-	return aligned_alloc(alignment, size);
+#elif defined(OQS_HAVE_MEMALIGN)
+	return memalign(alignment, size);
+#elif defined(__MINGW32__) || defined(__MINGW64__)
+	return __mingw_aligned_malloc(size, alignment);
+#elif defined(_MSC_VER)
+	return _aligned_malloc(size, alignment);
+#else
+	if (!size) {
```

Comment:
Thanks Sebastian! Would you mind adding a few comments here and in the corresponding part of `OQS_MEM_aligned_free` explaining what's going on with the pointer arithmetic?  Just so we have a record in case someone else needs to debug a future problem here.

Reply:
Done.